tools/grammars/compiler/converter.go

```diff
@@ -75,13 +75,35 @@ func (conv *Converter) AddGrammar(source string) error {
 	repo.FixRules(knownScopes)
 
 	if len(repo.Errors) > 0 {
-		fmt.Fprintf(os.Stderr, "The new grammar %s contains %d errors:\n",
-			repo, len(repo.Errors))
-		for _, err := range repo.Errors {
-			fmt.Fprintf(os.Stderr, "    - %s\n", err)
+		// Split out warnings from errors
+		warnings := []error{}
+		errors := []error{}
+		for _, e := range repo.Errors {
+			switch e.(type) {
+			case *MissingIncludeError, *UnknownKeysError:
+				warnings = append(warnings, e)
+			default:
+				errors = append(errors, e)
+			}
+		}
+		if len(errors) > 0 {
+			fmt.Fprintf(os.Stderr, "The new grammar %s contains %d errors:\n",
+				repo, len(errors))
+			for _, err := range errors {
+				fmt.Fprintf(os.Stderr, "    - %s\n", err)
```

Comment:
Personally, I'm not a fan of indenting error messages, as it takes up valuable space in what could be a narrow terminal window. I'd use the format

~~~
[filename]:[line-number]: [problematic-key]: [problem description]
~~~

~~~console
syntaxes/lean-markdown.json:1720: include: scope `text.log` not found
syntaxes/lean-markdown.json:2150: lists: invalid key for field `Repository[lists].Patterns[0]…`
~~~

Reply:
I'm not a fan either, but deliberately didn't adjust the formatting to reduce the complexity of this PR. Happy to do so if you're happy with a more complex PR.

Reply:
Thing is, users will no doubt be inclined to forward legitimate mistakes to the grammar upstream. Being able to filter out messages by category (such as "missing" grammar scopes), whether visually or through `grep`, will inevitably spell the difference between _"tl;dr, probably okay tho"_ and an _"I should fix that typo"_.

The fact this format is self-sorting is an added bonus, too.

Reply:
Looking into it, I think this level of information is beyond the scope of this PR. It's going to need a fair amount of refactoring to cater for line number etc. I think we can still remove the indentation though.

Reply:
Fair enough.

Reply:
Indentation removed. I've also removed the checkbox from the individual errors in the main report